### PR TITLE
US028: job role filtering

### PR DESCRIPTION
**Pull request by amordzon**

### CODE REVIEW

File: src/main/java/org/example/controllers/JobRoleController.java [hunk 1]
```diff
@@ -65,4 +67,34 @@ public Response getAllJobRoles() {
...
+            @ApiResponse(code = INTERNAL_SERVER_ERROR, message = "getAllJobRoles failed, SQL Exception"),
+            @ApiResponse(code = NOT_FOUND, message = "getAllJobRoles failed, DoesNotExistException")
+    })
+    @Path("/filter")
+    public Response getFilteredJobRoles(final @BeanParam JobRoleFilteredRequest jobRoleFilteredRequest) {
+        LOGGER.info("Get filtered job roles request received");
+        try {
+            return Response.ok().entity(jobRoleService.getFilteredJobRoles(jobRoleFilteredRequest)).build();
+        } catch (SQLException e) {
+            LOGGER.error("getFilteredJobRoles failed, SQL Exception\n" + e.getMessage());
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/JobRoleController.java]: maybe to add some more verbose error? Like "getting of filtered job roles failed due to SQL exception" ? also applies to any other error message in this file

File: src/main/java/org/example/models/JobRoleFilteredRequest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            value = "Job role's name",
+            required = true,
+            example = "Delivery manager")
+    @QueryParam("roleName")
+    private String roleName;
+
+    @ApiModelProperty(
+            value = "Role's location, defined by ENUM value",
+            required = true,
+            example = "[Gdańsk]")
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/JobRoleFilteredRequest.java]: defined by ENUM value: I would advise Gdansk ;) How UK guys are supposed to write ń :D

File: src/test/java/org/example/services/JobRoleServicesTest.java [hunk 3]
```diff
@@ -28,32 +33,62 @@ public void jobRolesListClean() {
...
     public void getAllJobRoles_WhenDaoThrowsSQLException_ExpectSQLExceptionToBeThrown()
             throws SQLException, ResultSetException {
-        Mockito.when(jobRoleDao.getAllJobRoles()).thenThrow(SQLException.class);
+        when(jobRoleDao.getAllJobRoles()).thenThrow(SQLException.class);
         assertThrows(SQLException.class, () -> jobRoleService.getAllJobRoles());
     }
+
+    @Test
+    public void testGetFilteredJobRolesEmptyList() throws SQLException, ResultSetException {
+        // Given
```
Comment (Alicja-Kainos) [inline src/test/java/org/example/services/JobRoleServicesTest.java]: those Given When Then comment in here are overkill :)

File: src/main/java/org/example/controllers/JobRoleController.java [hunk 4]
```diff
@@ -65,4 +67,34 @@ public Response getAllJobRoles() {
...
+    @Path("/filter")
+    public Response getFilteredJobRoles(final @BeanParam JobRoleFilteredRequest jobRoleFilteredRequest) {
+        LOGGER.info("Get filtered job roles request received");
+        try {
+            return Response.ok().entity(jobRoleService.getFilteredJobRoles(jobRoleFilteredRequest)).build();
+        } catch (SQLException e) {
+            LOGGER.error("getting filtered job roles failed due to SQL exception\n" + e.getMessage());
+            return Response.serverError().build();
+        } catch (DoesNotExistException | NullPointerException e) {
+            LOGGER.error("getFilteredJobRoles failed, DoesNotExistException\n" + e.getMessage());
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/JobRoleController.java]: ten error też ;)

File: src/main/java/org/example/controllers/JobRoleController.java [hunk 5]
```diff
@@ -65,4 +67,34 @@ public Response getAllJobRoles() {
...
+            return Response.status(Response.Status.NOT_FOUND).entity(e.getMessage()).build();
+        } catch (ResultSetException e) {
+            LOGGER.error("getFilteredJobRoles failed, ResultSetException\n" + e.getMessage());
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/JobRoleController.java]: ten też ;)

**Review by Alicja-Kainos — APPROVED** [review]

**Review by DariuszMech — APPROVED** [review]

**Pull request merged**